model/app.model.js

```diff
@@ -7,15 +7,101 @@ const selectTopics = () => {
     });
 };
 
-const selectArticle = (id) => {
+const selectArticleById = (id) => {
     const query = "SELECT * FROM articles WHERE article_id = $1;";
     let params = [id];
     return db.query(query, params).then((result) => {
+        if (result.rows.length === 0){
+            return Promise.reject({
+                status: 404,
+                msg: "Article does not exist"
+                })
+            }
+            else {
         return result.rows;
+            }
     });
 };
 
+const selectArticles = () => {
+    return db
+        .query(
+            `SELECT 
+            articles.article_id,
+            articles.author,
+            articles.title,
+            articles.topic,
+            articles.created_at,
+            articles.votes,
+            articles.article_img_url,
+            COALESCE(SUM(comments.comment_id),0) as comments_count
+    FROM articles LEFT JOIN comments
+    ON articles.article_id = comments.article_id
+    GROUP BY  articles.article_id, articles.author, articles.title, articles.topic, articles.created_at,   articles.votes, articles.article_img_url
+    ORDER BY articles.created_at DESC;`
+        )
+        .then((result) => {
+            return result.rows;
+        });
+};
+
+const selectAllCommentsForArticle = (id) => {
+    const query = "SELECT * FROM articles WHERE article_id = $1;";
+    let params = [id];
+    return db.query(query, params).then((result) => {
+        if (result.rows.length === 0){
+            return Promise.reject({
+                status: 404,
+                msg: "Article does not exist"
+                })
+            }
+            else {
+                const commentsQuery = `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at ASC`
+                return db.query(commentsQuery, params).then((comRes) => {
+                    return comRes.rows
+                })
+            }
+    })
+}
+
+
+
+const insertComment = (id, username, body) => {
+    if (Number.isNaN(parseInt(id))){
```

Comment:
We don't actually need to do this check, as PSQL will throw an error if the ID is not a number.We also don't need to check username, as A psql error will be thrown if this is violated. If the body is empty also, an error will be throw as it won't have the required keys to do the insert